dist/access_check/MODULARITY.md

```diff
@@ -0,0 +1,44 @@
+# Rules for Modularity
+
+## Definition of a Module
+
+* Whatever resides in a subdirectory under `src/` is considered to be a module.
+
+* The module names are pre-configured as a list of subdirectories under `src/`, with some exclusions like `include`, `checksum`, `os*`, etc. These exclusions are because these directories don’t really contain modular code. If the module name, as deduced by any method, is not in this list, there is no associated module, and the content is exempt from modularity checks.
+
+* Modules can have aliases. `block_cache` refers to `blkcache`, or `connection` is the same as `conn`.
+
+## Modules Content and Visibility Rules
+
+### 1. File Name/Path-Based Rules
+
+1. The default module for the content of a file is defined by these rules:
+   1. If the file is not in the `include` directory, the module is the topmost subdirectory name after `src/`.
+   2. For files in `include/`, the module name is derived from the file name with `.h` and `_inline` stripped.
+
+2. If the file name contains `_private`, everything in that file is considered private by default. Otherwise, everything is public by default.
+
+### 2. Identifier Name-Based Rules
+
+1. If the name starts with the `__wt_` prefix, the content is considered public.
+2. If the name starts with the `__wti_` prefix, the content is private.
+3. If the `__wt_` or `__wti_` prefix is followed by a valid module name and an underscore, the content is considered part of that module.
```

Comment:
The rules are clear, but a few key clarifications - 

- It would help to specify that these rules apply to names of functions, structs, macros, struct fields, etc., so it’s clear that they cover all identifier types.
- It might be worth mentioning how these rules relate to the header file name - that if a file is private (e.g., using a _private.h suffix), these naming rules don't apply, as the file’s private status takes precedence.

Reply:
The line below does that. Let me know if it's not clear enough.